openmdao/core/group.py

```diff
@@ -1273,6 +1273,7 @@ def _solve_nonlinear(self):
         # TODO: It is pointless to run this ahead of non-iterative solvers.
         for sub in self.system_iter(recurse=True):
             if hasattr(sub, 'guess_nonlinear'):
+                self._transfer('nonlinear', 'fwd')
```

Comment:
You probably want to use `self._transfer('nonlinear', 'fwd', isub)` here
e.g., https://github.com/OpenMDAO/blue/blob/master/openmdao/solvers/nonlinear/nonlinear_block_gs.py#L37